js/ui/tag_box.js

```diff
@@ -864,24 +865,66 @@ const TagBox = SelectBox.inherit({
 
     _renderTags: function() {
         const d = new Deferred();
+        let plainDataUsed = false;
 
-        this._loadTagsData().always((items) => {
-            if(this._disposed) {
-                d.reject();
-                return;
+        if(this._shouldGetDataFromPlain(this._valuesToUpdate)) {
```

Comment:
Let's make the names `_shouldGetDataFromPlain` and `_getItemsFromPlain` more consistent, etc   `_shouldGetItemsFromPlain`